lib/player/player_test.go

```diff
@@ -0,0 +1,265 @@
+		})
+	}
+}
+
+func TestClose(t *testing.T) {
+	clk := clockwork.NewFakeClock()
+	p, err := player.New(&player.Config{
+		Clock:     clk,
+		SessionID: "test-session",
+		Streamer:  &simpleStreamer{count: 2, delay: 1000},
+	})
+	require.NoError(t, err)
+
+	require.NoError(t, p.Play())
+
+	clk.BlockUntil(1) // player is now waiting to emit event 0
+
+	// advance to next event (player will have emitted event 0
+	// and will be waiting to emit event 1)
+	clk.Advance(1001 * time.Millisecond)
+	clk.BlockUntil(1)
+	evt := <-p.C()
+	require.Equal(t, int64(0), evt.GetIndex())
+
+	require.NoError(t, p.Close())
+
+	// channel should have been closed
+	_, ok := <-p.C()
+	require.False(t, ok, "player channel should have been closed")
+}
+
+func TestSeekForward(t *testing.T) {
+	clk := clockwork.NewFakeClock()
+	p, err := player.New(&player.Config{
+		Clock:     clk,
+		SessionID: "test-session",
+		Streamer:  &simpleStreamer{count: 10, delay: 1000},
+	})
+	require.NoError(t, err)
+	require.NoError(t, p.Play())
+
+	clk.BlockUntil(1) // player is now waiting to emit event 0
+
+	// advance playback until right before the last event
+	p.SetPos(9001 * time.Millisecond)
+
+	// advance the clock to unblock the player
+	// (it should now spit out all but the last event in rapid succession)
+	clk.Advance(1001 * time.Millisecond)
+
+	ch := make(chan struct{})
+	go func() {
+		defer close(ch)
+		for evt := range p.C() {
+			t.Logf("got event %v (delay=%v)", evt.GetID(), evt.GetCode())
+		}
+	}()
+
+	clk.BlockUntil(1)
+	require.Equal(t, int64(9000), p.LastPlayed())
+
+	clk.Advance(1001 * time.Millisecond)
+	select {
+	case <-ch:
+	case <-time.After(3 * time.Second):
+		require.FailNow(t, "player hasn't closed in time")
+	}
+}
+
+func TestRewind(t *testing.T) {
+	t.Skip("not implemented") // TODO(zmb3)
+
+	clk := clockwork.NewFakeClock()
+	p, err := player.New(&player.Config{
+		Clock:     clk,
+		SessionID: "test-session",
+		Streamer:  &simpleStreamer{count: 10, delay: 1000},
+	})
+	require.NoError(t, err)
+	require.NoError(t, p.Play())
+
+	// play through 7 events at regular speed
+	for i := 0; i < 7; i++ {
+		clk.BlockUntil(1)                    // player is now waiting to emit event
+		clk.Advance(1000 * time.Millisecond) // unblock event
+		<-p.C()                              // read event
+	}
+
+	// now "rewind" to the point just prior to event index 3 (4000 ms into session)
+	clk.BlockUntil(1)
+	p.SetPos(3900 * time.Millisecond)
+
+	// when we advance the clock, we expect the following behavior:
+	// - event index 7 (which we were blocked on) comes out right away
+	// - playback restarts, events 0 through 2 are emitted immediately
+	// - event index 3 is emitted after another 100ms
+	clk.Advance(1000 * time.Millisecond)
+	require.Equal(t, int64(7), (<-p.C()).GetIndex())
+	require.Equal(t, int64(0), (<-p.C()).GetIndex())
+	require.Equal(t, int64(1), (<-p.C()).GetIndex())
+	require.Equal(t, int64(2), (<-p.C()).GetIndex())
```

Comment:
Can you please add comments to those? 
```suggestion
	require.Equal(t, int64(7), (<-p.C()).GetIndex(), "Expected the stream to process 7 events")
	require.Equal(t, int64(0), (<-p.C()).GetIndex(), "Expected to emit 0 events due to pause")
	require.Equal(t, int64(1), (<-p.C()).GetIndex(), "...")
	require.Equal(t, int64(2), (<-p.C()).GetIndex(), "...")
```